TestCluster for TestingSiloHost

amccool opened the pull request:
for #2232 
Moved to TestCluster.  Using defaults for ports
still more to go, but this is a start

jdom commented on test/TesterInternal/StreamingTests/HaloStreamSubscribeTests.cs:
```diff
@@ -11,23 +11,39 @@
+                options.ClusterConfiguration.Globals.UseMessageBatching = false;
+                options.ClusterConfiguration.Globals.MaxMessageBatchingSize = 100;
+                options.ClusterConfiguration.Globals.LivenessType = Orleans.Runtime.Configuration.GlobalConfiguration.LivenessProviderType.MembershipTableGrain;
+                options.ClusterConfiguration.Globals.ReminderServiceType = Orleans.Runtime.Configuration.GlobalConfiguration.ReminderServiceProviderType.ReminderTableGrain;
```
I believe these last 5 settings are not really needed

jdom commented:
@dotnet-bot test this please

amccool commented:
Getting errors from the Azure storage, looking for a DataConnectionString.   Should these tests be using
`DataConnectionString="UseDevelopmentStorage=true"`?

jdom commented:
If these are functionals, they would be getting a real connection string when run in our VSO. Some tests work fine with Azure storage emulator (are you running it?).
Nevertheless, at some point we should do a pass to see which tests rely on either the emulator or real azure storage, and skip the test if the environment is not correct (using SkippableFact).
For now I can run them here.

jdom commented:
@dotnet-bot test this please

amccool commented:
In this push, again, I tried to follow the xml configs and code config.  There appears to be some overlap, but not enough to start combining the test fixtures.

amccool commented:
last commit is breaking.
`ShardedStorageProvider` needs it Children populated by existing `StorageProviders`, which is not possible

`StatisticsProviders` need a fleshed out `RegisterStatisticsProvider<T>`.  I added the method, but the implementation is not complete.

amccool commented:
@jdom I reverted that last two commits.   It should match with your last functional test. What should I do with the two commits I reverted out of this PR?

jdom commented:
I'd say create a separate PR later on, after this one gets merged.
BTW, there has been a few moves lately in our test projects, so now there are conflicts. Are you able to do a rebase on top of the current master and fix the conflicts? I could help you if you are not very used to git.

amccool commented:
ok - I will wait on the next two commits, and submit them separately.
so - rebase against master, and push to this PR.   I think I can handle that.  I'll let you know if I need some assistance.

jdom commented:
BTW, in case someone else is reviewing, we are troubleshooting this with @amccool directly, as there are some failing functional tests. Please do not merge yet

jdom commented:
Thanks for the update, kicked off another round of testing

jdom commented:
Ran the tests, and as you mentioned in that commit, PersistenceGrainTest still has issues (looks like they are the only remaining issues). Can you look into it or do you need help?

jdom commented:
BTW, took a quick look and already solved it. I'll push to your branch, as I think I still have access.

jdom commented:
Thanks @amccool ! :)
